storage/aptosdb/src/pruner/worker.rs

```diff
@@ -111,15 +115,20 @@ impl Worker {
                 // On `Command::Quit` inform the outer loop to quit by returning `false`.
                 Command::Quit => return false,
                 Command::Prune { target_db_versions } => {
-                    for (new_target_version, pruner) in
+                    for (new_target_version_option, pruner_option) in
                         zip_eq(&target_db_versions, &self.db_pruners)
                     {
-                        if *new_target_version > pruner.lock().target_version() {
-                            // Switch to non-blocking to allow some work to be done after the
-                            // channel has drained.
-                            self.blocking_recv = false;
+                        if let Some(pruner) = pruner_option {
+                            assert!(new_target_version_option.is_some());
+                            if let Some(new_target_version) = new_target_version_option {
```

Comment:
nit: instead of `if let`, you can just `unwrap` and avoid an extra level of nesting.